src/charmed_kubeflow_chisme/kubernetes/_kubernetes_resource_handler.py

```diff
@@ -152,7 +152,13 @@ def delete(self):
         )
 
         resources_to_delete = self.get_deployed_resources()
-        delete_many(self.lightkube_client, resources_to_delete)
+        try:
+            delete_many(self.lightkube_client, resources_to_delete)
+        except ApiError as error:
```

Comment:
I wonder if this may give a false sense of security.  Say we try to `delete_many` these resources:
* `ResourceA`: does not exist
* `ResourceB`: does exist

`delete_many` will raise on `ResourceA` and never attempt to delete `ResourceB`, so KRH.delete() will act as if it worked but really it didn't.  I don't think we want to suppress this sort of situation.

Would it be better to add this logic in `delete_many`?  Maybe with a option like `ignore_missing=True`?  That way `delete_many` doesn't exit early in this case.  

Or should `delete_many` be more persistent on all errors, not just 404's?  Maybe it should:
* try to delete all resources, regardless of errors
* (optionally) ignore (but log) 404 errors
* if a single error is encountered, raise it at the end.  If multiple are encountered, reraise one with a message saying there's multiple errors and the rest were logged

?  Maybe that's going too deep

Reply:
> Would it be better to add this logic in `delete_many`? Maybe with a option like `ignore_missing=True`? That way `delete_many` doesn't exit early in this case.

I totally agree with this one!

Reply:
> Or should `delete_many` be more persistent on all errors, not just 404's?

I'd like to think a bit more about this. While ignoring a 404 is a common practice when deleting resources, ignoring everything feels unintuitive. We could discuss about different errors that could be raised on deletion and whether it would make sense to ignore them. For instance, if a 403 or 405 is raised, we probably don't want to keep trying for the rest of the resources (i.e. probably pointless since we're handling the same kind of resources in the same namespace). If it's a 400 or a 409 it could mean that we're on the verge of performing unwanted actions.

Reply:
I am leaning toward `delete_many`:
* trying to delete everything, regardless of whether some raise exceptions
* raising all errors at the end (or really raising 1 and logging/saying there were others)
* optionally suppressing 404's

Maybe like:

```python
def delete_many(..., ignore_missing=True):
  handled_exceptions = []
  for resource in resources:
    try:
      lightkube.delete(resource)
    except ApiError as err:
      if err==404 and ignore_missing=True:
        log.info(...)
        continue
        handled_exceptions.append(err)
  if len(handled_exceptions) == 1:
    raise SomeError("Caught 1 exception when deleting resources") from handled_exceptions[0]
  elif len(handled_exceptions) > 1:
    raise SomeError("Caught {len(handled_exceptions)} exceptions when deleting resources.  Reraised the first, all errors are: {handled_exceptions}") from handled_exceptions[0]
```

the error raising at the end is kinda messy and needs better choices, but still :D

Reply:
This changes the default behaviour of delete_many though (previously it would raise on the first 404, now it ignores all 404's).  Normally I'd be against changing the default behaviour or something, but I think any time we use `delete_many` right now what we'd prefer to use is `delete_many(..., ignore_missing=True)`.  And since ignoring failure to delete an object that doesn't exist isn't so problematic, I'm ok with it.  its debatable though

Reply:
Note that we can wrap all handled exceptions in a single Exception object:
```python
raise Exception(f"Caught {len(handled_exceptions)} exceptions while deleting resources", handled_exceptions)
```

Reply:
on the error raising, we dont need to make it complicated.  Could be that we just do:
```python
if len(handled_exceptions) > 0:
  raise SomeError(f"Caught {len(handled_exceptions)} errors while deleting resources - reraised the first.")
```

I know the first time I get this I'm gonna say "what were the other errors????" but if I'm in the minority here I can deal :D